src/mmw/apps/modeling/tasks.py

```diff
@@ -146,8 +146,19 @@ def analyze_pointsource(area_of_interest):
     return [point_source_pollution(area_of_interest)]
 
 
+def ensure_aoi_4d(area_of_interest):
+    # In rare instances, this method receives a five-dimensional array for these
+    # coordinates. This check is here to verify whether the `area_of_interest` is
+    # four- or five-dimensional, then assigns `pairs` appropriately.
+    if type(area_of_interest['coordinates'][0][0][0][0]) is list:
```

Comment:
This level of nesting indicates that the geometry that the RWD service is producing is a `MultiPolygon` instead of a `Polygon` feature.  Perhaps a more readable check would be for `area_of_interest['type'] == 'MultiPolygon'`.  That said, a MultiPolygon has a list of coordinates at that dimension because it is comprised of more than one full polygon and I suppose that taking just the first ring of one is not quite as accurate for computing an average latitude.  However, since the result is just an approximation, it's probably not a big deal.